### PR TITLE
feat: add additional env values

### PR DESCRIPTION
**Pull request by nicovak**
## What was changed

Ability to have additional environment variable in deployment like so : 

- Globally for server (will be applied to all service deployment)

```yaml
server:
  additionalEnv:
    - name: "HELLO"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: world
    - name: "WORLD"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: hello
```

- For specific services only

```yaml
server:
  frontend:
    additionalEnv:
      - name: "HELLO"
        valueFrom:
          secretKeyRef:
            name: temporal-default-store
            key: world
      - name: "WORLD"
        valueFrom:
          secretKeyRef:
            name: temporal-default-store
            key: hello
```

- For global server + service override

```yaml
server:
  additionalEnv:
    - name: "HELLO"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: world
    - name: "WORLD"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: hello

  history:
    additionalEnv:
      - name: "HELLO_OVERRIDE"
        valueFrom:
          secretKeyRef:
            name: temporal-default-store
            key: world
      - name: "WORLD_OVERRIDE"
        valueFrom:
          secretKeyRef:
            name: temporal-default-store
            key: hello
```

- For web deployment

```yaml
web:
  additionalEnv:
    - name: "HELLO"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: world
    - name: "WORLD"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: hello
```

- For admintools deployment

```yaml
admintools:
  additionalEnv:
    - name: "HELLO"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: world
    - name: "WORLD"
      valueFrom:
        secretKeyRef:
          name: temporal-default-store
          key: hello
```

## Why?
It's a commonly asked feature to add additional environment variables to deployments

## Checklist

1. Closes
No

3. How was this tested:
Locally with multiple additional environment variables

4. Any docs updates needed?
No

### CODE REVIEW

File: templates/server-deployment.yaml [hunk 1]
```diff
@@ -117,6 +117,7 @@ spec:
             - name: TEMPORAL_VERSION_CHECK_DISABLED
               value: "1"
           {{- end }}
+          {{- include "additional-env" (default $.Values.server.additionalEnv $serviceValues.additionalEnv) | indent 10 }}
```
Comment (underrun) [inline templates/server-deployment.yaml:120]: why is `additional-env` needed? why not handle this the same way that `additionalVolumeMounts` is handled? If there is a reason why the `additional-env` approach is better could we replace all the other handling for lists of additional things?

**Review by underrun — CHANGES_REQUESTED** [review]

File: templates/server-deployment.yaml (hunk 1 above)
Comment (nicovak) [reply templates/server-deployment.yaml]: You are right. I pushed the rebase and added more cases in PR description.

**Comment by nicovak** [comment]
@underrun any updates on this ?

**Comment by medamineamara** [comment]
I need this feature
any updates ?

**Comment by medamineamara** [comment]
I have been working with this feature and it's working fine as expected

**Review by mindaugasrukas — APPROVED** [review]
LGTM

**Pull request merged**